subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -689,4 +681,131 @@ static struct lll_df_adv_cfg *df_adv_cfg_acquire(void)
 
 	return df_adv_cfg;
 }
+
+static uint8_t set_cte_info_in_per_adv_chain(struct ll_adv_set *adv, struct lll_df_adv_cfg *df_cfg,
+					     uint8_t *ter_idx)
+{
+	void *extra_data_prev, *extra_data;
+	struct adv_pdu_field_data pdu_data;
+	struct pdu_adv *pdu_prev, *pdu;
+	struct lll_adv_sync *lll_sync;
+	struct pdu_cte_info cte_info;
+	int err;
+
+	lll_sync = adv->lll.sync;
+
+	cte_info.type = df_cfg->cte_type;
+	cte_info.time = df_cfg->cte_length;
+	pdu_data.field_data = (uint8_t *)&cte_info;
+	pdu_data.extra_data = df_cfg;
+
+	/* Note: ULL_ADV_PDU_HDR_FIELD_CTE_INFO is just information that extra_data
+	 * is required in case of this ull_adv_sync_pdu_alloc.
+	 * Other flags here do not change anything. It may be changed to use
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0, &pdu_data, &pdu_prev,
+				     &pdu, &extra_data_prev, &extra_data, ter_idx);
+	if (err) {
+		return err;
+	}
+
+	if (extra_data) {
+		ull_adv_sync_extra_data_set_clear(extra_data_prev,
+						  extra_data,
+						  ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+						  0, &pdu_data);
+	}
```

Comment:
If there is no `extra_data` do we need to clear the previous data?

Reply:
Actually `extra_data_prev` is not needed here because is it not used. Every time CTE is set in a chain new configuration is set also. Besides that `ull_adv_sync_pdu_alloc` and `ull_adv_sync_pdu_set_clear` should be updated because they has unnecessary parameters.